### ISSUE TITLE
Figure out a way to re-produce a bootable ISO

### ISSUE THREAD

**Issue by minexew** [minexew/TempleOS#2]
The closest I got was using

`mkisofs -J -r -l -b 0000Boot/0000Kernel.BIN.C -no-emul-boot -o Temple.iso TempleOSCD/`

The kernel gets loaded, but fails to identify the boot drive, asks for the I/O port number and then crashes.
Perhaps not all of the kernel gets loaded. (tested on QEMU)

What the built-in `ISO9660ISO` function does to produce a bootable image, is use a small Stage-1 loader and binary-patch in the kernel's raw on-disc location. Which is, needless to say, pretty clunky.

**Comment by grepwood** [comment]
Did you supply the port numbers? Terry said in the readme that you need to do that because TempleOS can't figure those out.

**Comment by digitalcombine** [comment]
I don't believe you can create an ISO this way for TempleOS. TempleOS don't seem to understand the standard ISO9660 filesystems and uses it's own RedSea ISO filesystem instead.

To roll your own ISO, first you install TempleOS on to a hard drive file and run it from something like qemu. In the T:/Misc directory is DoDistro.HC.Z which will create an ISO for you. From there you can retrieve the ISO with his qemu scripts (I believe, but not 100% sure).

**Comment by mczero80** [comment]
I am too interested in building a new ISO file. I watched a Terry video on youtube where he instructs how to build a new distro, but the starting ISO that I downloaded was missing the Kernel/KEnd.HC that he was editing in his video. This repository has that file... Perhaps I am thinking to complicated. I don't know.

Sorry to disturb this issue thread

**Comment by minexew** [comment]
@mczero80 Try to find an ISO with SHA-1 hash 1a1ec79...

It might be in https://archive.org/details/TempleOS_ISO_Archive as TempleOSV503, I don't remember

**Comment by mczero80** [comment]
> @mczero80 Try to find an ISO with SHA-1 hash 1a1ec79...
> 
> It might be in https://archive.org/details/TempleOS_ISO_Archive as TempleOSV503, I don't remember

That's the correct ISO! Thank you very much :+1:

**Comment by omerien** [comment]
GOT THE ISO TO WORK
I'm using UTM (it's for Mac and it's based on QEMU)
Here's my configuration:
<img width="830" alt="Capture d’écran 2022-08-11 à 15 42 39" src="https://user-images.githubusercontent.com/53191414/184147555-3a483a42-c27f-42d9-9991-c82e0c2b864d.png">
<img width="714" alt="Capture d’écran 2022-08-11 à 15 42 57" src="https://user-images.githubusercontent.com/53191414/184147565-32b32307-f655-4492-978b-7d1681022d13.png">
<img width="838" alt="Capture d’écran 2022-08-11 à 15 43 20" src="https://user-images.githubusercontent.com/53191414/184147583-6907dd1c-428a-4546-bafb-8cce6ae720d1.png">

**Comment by omerien** [comment]
(I'm talking about the ISO you created, and yes I had the same message as you before changing randomly some settings)

**Comment by Ashveil1** [comment]
When you know holyC#

**Comment by Ashveil1** [comment]
I doing kernel temple OS for Intel i5 gen 6 and Intel UHD graphics
